internal/core/window.rs

```diff
@@ -321,6 +324,23 @@ impl WindowInner {
     pub fn process_mouse_input(&self, mut event: MouseEvent) {
         crate::animations::update_animations();
 
+        // handle double click
+        if let MouseEvent::Pressed { position, .. } = event {
+            if let Some(pressed_time_stamp) = self.pressed_time_stamp.get() {
+                if crate::animations::Instant::now() - pressed_time_stamp
+                    < crate::platform::PLATFORM_INSTANCE
+                        .with(|p| p.get().map(|p| p.double_click_interval()))
+                        .unwrap_or_default()
+                {
+                    self.process_mouse_input(MouseEvent::DoubleClicked { position });
+                }
+
+                self.pressed_time_stamp.set(None);
```

Comment:
I wonder if this simplification compiles?

```suggestion
            if let Some(pressed_time_stamp) = self.pressed_time_stamp.take() {
                if crate::animations::Instant::now() - pressed_time_stamp
                    < crate::platform::PLATFORM_INSTANCE
                        .with(|p| p.get().map(|p| p.double_click_interval()))
                        .unwrap_or_default()
                {
                    self.process_mouse_input(MouseEvent::DoubleClicked { position });
                }
```